### ISSUE TITLE
no autocomplete results for defined macros inside function/variable definition/declaration

### ISSUE THREAD

**Issue by Dromantor** [microsoft/vscode-cpptools#8973]
I guess this is somewhat by design that no autocomplete results except snippets are shown inside a declaration because you obviously want to declare something new.
![grafik](https://user-images.githubusercontent.com/1822013/156762365-1aa1e088-bef2-4176-9060-c2a46e9023f5.png)
But for the same reason that snippets are shown (I guess because the snippet could hold something like a default function you may want to use) it would make sense to also show macros. In my case i have some #define-s for a collection of function and variable attributes like these for interrupt routines:
![grafik](https://user-images.githubusercontent.com/1822013/156762825-35ac7fdc-7ac4-4105-b527-322deb25d8e1.png)
and it would be great if those would also show up. 
I also sometimes have #define-aliased the names of interrupt routines and hardware registers so I can easily swap them out later without digging through the code.
I would really like if there was at least an option to also show macros.

**Comment by AdhTri001** [comment]
I opened the issue tracker of this repo because I was facing the same issue, and I thought new update might have some bug because extension used to have autocomplete. Then the following showup in the output panel of the vs-code when I reloaded it:

```
For C source files, IntelliSenseMode was changed from "windows-clang-x64" to "windows-gcc-x64" based on compiler args and querying compilerPath: "D:/mingw/bin/clang++.exe"
For C++ source files, IntelliSenseMode was changed from "windows-clang-x64" to "windows-gcc-x64" based on compiler args and querying compilerPath: "D:/mingw/bin/clang++.exe"
```

I don't know if this is the only reason why I was facing that issue but looks like mine is fixed might help you too.

**Comment by Dromantor** [comment]
I just tested it with a minimal test file in a new workspace with different compiler configurations, all the same.
![grafik](https://user-images.githubusercontent.com/1822013/156776894-9134472c-a450-443c-b1a9-80ea3fb07429.png)
No intellisense warnings in my output. Would have been great if it was that easy :)

**Comment by sean-mcmanus** [comment]
This will work as long as the type is not a predefined keyword (or some other special cases), although it'll show more than just macros. However, I believe it should be fixed to show completions anyway when it's manually invoked (such as by Ctrl+Space), which we currently do after a struct/class/etc. keyword. Also, the inconsistency based on the type of the previous token is strange -- I think we originally did that to match VS but VS no longer seems to behave that way (i.e. they disable completions even if the token on the left is a user defined identifier).

We don't normally who only a specific type of completion, but I suppose we could in this case.

**Comment by Colengms** [comment]
Hi @AdhTri001 .  Could you refer me to where/how you acquired the (mingw?) version of clang that you are using that appears to be getting misidentified as gcc?  I'd like to investigate that further.

**Comment by AdhTri001** [comment]
@Colengms [brechtsanders/winlibs_mingw](https://github.com/brechtsanders/winlibs_mingw) gives builds latest version of mingw and release it.

**Comment by Colengms** [comment]
Hi @AdhTri001 .  The info at that link refers to gcc (MinGW is gcc/g++), and does not refer to clang.  But, the compiler path you specified points to an instance of clang++.  (`D:/mingw/bin/clang++.exe`).  I don't see a clang++.exe in the archives provided by that repo.  If that executable is actually a renamed (or symlink to) gcc/g++, the configuration warning would be correct.

**Comment by AdhTri001** [comment]
No It isn't rename. To get llvm stuff with mingw, you have to download "GCC 11.2.0 + LLVM/Clang/LLD/LLDB 13.0.0 + MinGW-w64 9.0.0 (UCRT) - release 3" that is: [brechtsanders/winlibs_mingw/releases/tag/11.2.0-9.0.0-ucrt-r3](https://github.com/brechtsanders/winlibs_mingw/releases/tag/11.2.0-9.0.0-ucrt-r3) this release. Or simply check website winlibs.com

**Comment by AdhTri001** [comment]
@Colengms did you managed to find the source of problem? I still recieve the same output whenever I open vs code.

**Comment by Colengms** [comment]
Hi @AdhTri001 .  I created a new issue: https://github.com/microsoft/vscode-cpptools/issues/9024

I have a fix which will hopefully make it into 1.9.5.

**Comment by sean-mcmanus** [comment]
@AdhTri001 The fix for your issue should be in 1.9.4.

**Comment by sean-mcmanus** [comment]
@AdhTri001 However, it got reverted in 1.9.5 due to a different issue. It's planned to be added back in 1.9.6.

**Comment by sean-mcmanus** [comment]
This should be fixed with 1.10.2 (pre-release).
